src/api/metadata.rs

```diff
@@ -17,133 +20,36 @@ pub struct TypeMeta {
     pub kind: Option<String>,
 }
 
-
-#[derive(Deserialize, Serialize, Clone, Default)]
-pub struct ListMeta {
-    pub continue_: Option<String>,
-    pub resourceVersion: Option<String>,
-    pub selfLink: Option<String>,
+pub trait MetaContent: Metadata {
+    fn resource_ver(&self) -> Option<String>;
+    fn name(&self) -> String;
+    fn namespace(&self) -> Option<String>;
```

Comment:
Very minor nitpick, but.. I think all of these could borrow the `String` instead?

Reply:
Yeah, can do. Was flipping between when implementing it. Switching now. Want to rename the trait to something shorter anyway. Maybe just straight `Meta`.

Reply:
:+1:

Reply:
Actually, I might still clone in the non-expensive getters like `Meta::name` , because it's really conflicting in all the examples where we need to burrow multiple parts from the main object:

```rust
        let name = Meta::name(&pod);
        let phase = pod.status.as_ref().unwrap().phase.as_ref().unwrap();
        let containers = pod.spec.as_ref().unwrap()
                .containers
                .into_iter()
                .map(|c| c.name)
                .collect::<Vec<_>>();
        info!("Found initial pod {} ({}) with {:?}", name, phase, containers);
```

If we all burrowed from pod in `Meta`, then the least significant burrow for the name would prevent us from burrowing pod.spec  (cannot move out of shared reference).

Reply:
The client code already knows what `Metadata::Ty` will be, so it should just be able to access the field directly (`let meta = &pod.metadata.name;`). Then again, I can see that it would be a pain to change all the examples again.. :/

Reply:
Once you do that  you quickly hit `move out of burrowed context` since metadata is optional. I was trying a bunch of different configurations, and have gone for the most ergonomic one so far,  maybe there is a better one, but also kind of just want to get this done now :D

Reply:
Fair enough!